src/Source.cpp

```diff
@@ -10,16 +11,26 @@ SourcePtr Source::create(List spec) {
   std::string subclass(as<CharacterVector>(spec.attr("class"))[0]);
 
   int skip = as<int>(spec["skip"]);
-  std::string comment = as<std::string>(spec["comment"]);
+  SEXP comment_sexp = spec["comment"];
```

Comment:
Well you never actually check that `comment_sexp` is a character vector and I'm not sure `STRING_ELT(comment_sexp, i)` checks its inputs. So you at least need a test for typeof.